backup-stores/azure/src/main/java/io/camunda/zeebe/backup/azure/ManifestManager.java

```diff
@@ -0,0 +1,112 @@
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.backup.azure;
+
+import static com.fasterxml.jackson.databind.SerializationFeature.WRITE_DATES_AS_TIMESTAMPS;
+
+import com.azure.core.util.BinaryData;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.azure.storage.blob.specialized.BlockBlobClient;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import io.camunda.zeebe.backup.api.Backup;
+import io.camunda.zeebe.backup.azure.AzureBackupStoreException.UnexpectedManifestState;
+import io.camunda.zeebe.backup.azure.manifest.Manifest;
+import io.camunda.zeebe.backup.azure.manifest.Manifest.InProgressManifest;
+
+public final class ManifestManager {
+  /**
+   * The path format consists of the following elements:
+   *
+   * <ul>
+   *   <li>{@code "manifests"}
+   *   <li>{@code partitionId}
+   *   <li>{@code checkpointId}
+   *   <li>{@code nodeId}
+   *   <li>{@code "manifest.json"}
+   * </ul>
+   *
+   * The path format is constructed by partitionId/checkpointId/nodeId/manifest.json
+   */
+  private static final String MANIFEST_PATH_FORMAT = "%s/%s/%s/manifest.json";
+
+  private static final ObjectMapper MAPPER =
+      new ObjectMapper()
+          .registerModule(new Jdk8Module())
+          .registerModule(new JavaTimeModule())
+          .disable(WRITE_DATES_AS_TIMESTAMPS)
+          .setSerializationInclusion(Include.NON_ABSENT);
+  private boolean containerCreated = false;
+  private final BlobContainerClient blobContainerClient;
+
+  ManifestManager(final BlobContainerClient blobContainerClient) {
+    this.blobContainerClient = blobContainerClient;
+  }
+
+  InProgressManifest createInitialManifest(final Backup backup) {
+    try {
+      final var manifest = Manifest.createInProgress(backup);
+      final byte[] serializedManifest;
+      if (!containerCreated) {
+        blobContainerClient.createIfNotExists();
+        containerCreated = true;
+      }
+      serializedManifest = MAPPER.writeValueAsBytes(manifest);
+      final BlockBlobClient blobClient =
+          blobContainerClient.getBlobClient(manifestPath((manifest))).getBlockBlobClient();
+      blobClient.upload(BinaryData.fromBytes(serializedManifest));
+      return manifest;
+    } catch (final JsonProcessingException e) {
+      throw new RuntimeException(e);
+    } catch (final BlobStorageException e) {
+      throw new UnexpectedManifestState(e.getMessage());
```

Comment:
The error codes that `BlobStorageException` can throw are in this [page](https://learn.microsoft.com/en-us/java/api/com.azure.storage.blob.models.bloberrorcode?view=azure-java-stable)
I think the ones that more sense for us to throw `UnexpectedManifestState ` are probably:
```
BlobErrorCode.BLOB_NOT_FOUND,
BlobErrorCode.BLOB_OVERWRITTEN,
BlobErrorCode.BLOB_ARCHIVED,
BlobErrorCode.BLOB_NOT_ARCHIVED,
BlobErrorCode.BLOB_BEING_REHYDRATED
```
wdyt? Should we add any other? @oleschoenburg

Reply:
Hm, I think none of this really map cleanly to `UnexpectedManifestState`. If you compare this exception with `BackupInInvalidStateException` from s3 and `UnexpectedManifestState` from gcs, they are only thrown for semantic errors such as updating a manifest to completed but there's no in-progress manifest. Another example is the gcs store throwing this when the manifest was modified unexpectedly.

As far as I can see, the azure implementation so far doesn't check anything like this -> we don't really have a reason to throw `UnexpectedManifestState` yet. In my opinion this highlights a gap in the azure store and we should make at least some semantic checks, perhaps similar to how the GCS store does it.

Reply:
is it worth checking the manifest state before we manipulate it? And in the case it is not the expected one, we'd throw the exception.

Reply:
There appear to be some lease strategies per blob. This might be interesting, as it would restrict other uploads during reading (verifying state) and then writing/overwriting.